hvcc/generators/ir2c/static/HvControlExpr.h

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2022-2025 Daniel Billotte, Wasted Audio
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+ * REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+ * INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+ * LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+ * OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+ * PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#ifndef _HEAVY_CONTROL_EXPR_H_
+#define _HEAVY_CONTROL_EXPR_H_
+
+#include "HvHeavyInternal.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define MAX_EXPR_ARGS 100
+
+typedef struct ControlExpr {
+  float args[MAX_EXPR_ARGS];
+  float(*eval_fptr)(const float*);
+} ControlExpr;
+
+hv_size_t cExpr_init(ControlExpr *o, float(*eval_fptr)(const float*));
+
+void cExpr_free(ControlExpr *o);
+
+void cExpr_onMessage(HeavyContextInterface *_c, ControlExpr *o, int letIn, const HvMessage *m,
+    void (*sendMessage)(HeavyContextInterface *, int, const HvMessage *));
+
+static inline float expr_modf(float mod)
```

Comment:
@grrrr I'm thinking that we should move these to `HvUtils.h` what do you think would be the best form between `static inline`  vs `#define`?

Reply:
i would definitely go for static inline. The factorial is an edge case though, i think i would not necessarily inline code with non-constant loop counts.

Reply:
@grrrr so I was looking at where we are with `expr~` and was thinking of doing this: https://github.com/Wasted-Audio/hvcc/pull/277/files

So we refer to the same aliased set of operations, using standard naming convention.
This way we can also reuse the "naive" single value processing to be used for the missing objects in `exp~` (on embedded we don't have SIMD anyway)

Reply:
looks good!